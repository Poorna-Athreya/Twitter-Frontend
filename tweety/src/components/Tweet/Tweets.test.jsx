/* eslint-disable max-len */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Tweets from './index';

describe('Tweets', () => {
  it('should take a snapshot for Tweets', () => {
    const mockTweetText = 'Great Project!';
    const mockTweetCreatedAt = '2022-03-23T16:39:08.227Z';
    const { asFragment } = render(<Tweets text={mockTweetText} createdAt={mockTweetCreatedAt} />);
    expect(asFragment(<Tweets text={mockTweetText} createdAt={mockTweetCreatedAt} />)).toMatchSnapshot();
  });
  it('should have the required elements when Tweets is rendered', () => {
    const mockTweetText = 'Great Project!';
    const mockTweetCreatedAt = '2022-03-23T16:39:08.227Z';
    render(<Tweets text={mockTweetText} createdAt={mockTweetCreatedAt} />);
    expect(screen.queryByTestId('tweetText')).toBeTruthy();
    expect(screen.getByText(mockTweetText)).toBeTruthy();
    expect(screen.queryByTestId('tweetCreatedAt')).toBeTruthy();
    expect(screen.getByText(mockTweetCreatedAt)).toBeTruthy();
  });
});
