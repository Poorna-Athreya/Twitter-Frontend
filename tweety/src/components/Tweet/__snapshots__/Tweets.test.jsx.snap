// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweets should take a snapshot for Tweets 1`] = `
<DocumentFragment>
  <div
    class="tweet"
  >
    <h3
      class="tweet-text"
      data-testid="tweetText"
    >
      Great Project!
    </h3>
    <p
      class="tweet-timestamp"
      data-testid="tweetCreatedAt"
    >
      2022-03-23T16:39:08.227Z
    </p>
  </div>
</DocumentFragment>
`;
